tests/test_dpm.py

```diff
@@ -0,0 +1,655 @@
+                     ["C", "A", "C", "D"]]
+
+        attributes = [[5, 5, 3, 8, 2],
+                      [1, 3, 3],
+                      [4, 5, 2, 1]]
+
+        seq2pat = Seq2Pat(sequences=sequences)
+
+        # Create price attributes
+        price = Attribute(values=attributes)
+
+        price_ct = price.median() <= 2
+
+        patterns = [['B', 'D', 1], ['A', 'C', 'D', 1]]
+
+        # Create encoding
+        with self.assertRaises(ValueError):
+            # This should fail when patterns have frequency appended to the end of patterns
+            encodings = get_one_hot_encodings(sequences, patterns, constraints=[price_ct], drop_pattern_frequency=False)
+
+    def test_constraints_as_positional_arg(self):
+        sequences = [["A", "A", "B", "A", "D"],
+                     ["C", "B", "A"],
+                     ["C", "A", "C", "D"]]
+
+        values = [[5, 5, 3, 8, 2],
+                  [1, 3, 3],
+                  [4, 5, 2, 1]]
+
+        # Seq2Pat over 3 sequences
+        seq2pat = Seq2Pat(sequences)
+
+        # Create price attributes
+        price = Attribute(values)
+        # Create price attributes
+        price_ct = 3 <= price.median() <= 4
+        # Constraint to specify the median of prices in a pattern
+        seq2pat.add_constraint(price_ct)
+        # Find sequences with min_frequency=2
+        patterns = seq2pat.get_patterns(min_frequency=2)
+
+        # Create encoding with/without constraints
+        encodings = get_one_hot_encodings(sequences, patterns, [price_ct], drop_pattern_frequency=True)
+
+        self.assertListEqual([[1], [0], [1]], encodings.values[:, 1:].tolist())
+
+    def test_dichotomic_pattern_mining_union(self):
+        sequences_pos = [["A", "A", "B", "A", "D"],
+                         ["C", "B", "A"],
+                         ["C", "A", "C", "D"]]
+
+        # update attributes in the 2nd and 3rd row to add ['C', 'A']
+        values_pos = [[5, 5, 3, 8, 2],
+                      [3, 3, 3],
+                      [3, 5, 2, 1]]
+
+        sequences_neg = [["A", "A", "B", "A", "D"],
+                         ["C", "B", "A"],
+                         ["C", "A", "C", "D"]]
+
+        # Updating attributes in the first row to add ['B', 'A']
+        values_neg = [[5, 5, 3, 3, 2],
+                      [1, 3, 3],
+                      [4, 5, 2, 1]]
+
+        # Define seq2pat_pos for positive cohort
+        seq2pat_pos = Seq2Pat(sequences=sequences_pos)
+        price_attr_pos = Attribute(values=values_pos)
+        price_attr_ct_pos = 3 <= price_attr_pos.median() <= 4
+        seq2pat_pos.add_constraint(price_attr_ct_pos)
+
+        # Define seq2pat_neg for negative cohort
+        seq2pat_neg = Seq2Pat(sequences=sequences_neg)
+        price_attr_neg = Attribute(values=values_neg)
+        price_attr_ct_neg = 3 <= price_attr_neg.median() <= 4
+        seq2pat_neg.add_constraint(price_attr_ct_neg)
+
+        sequences = sequences_pos + sequences_neg
```

Comment:
Sorry for the confusion. I am concatenating the two lists of sequences, not the seq2pat objects. But actually these should be used in encoding rather than in the test for DPM patterns, so these can be dropped.